### ISSUE TITLE
NAT confuses shoop

### ISSUE THREAD

**Issue by hukka** [mcginty/shoop#24]
- opening UDT connection...[E] errrrrrrr connecting to 192.168.1.4:60500 - UdtError { err_code: 1001, err_msg: "Connection setup failure: connection time out." }

which makes sense, as the client is not inside the LAN. But it should make the UDT connection to the same ip as the original ssh did.

**Comment by mcginty** [comment]
Hmm, I guess that means we can't trust the SSH_CONNECTION env variable? I think that's how mosh does it though, I wonder what's different about your situation.

**Comment by hukka** [comment]
Indeed, when connecting from NATed client to NATed server, the SSH_CONNECTION shows the client's external IP on the server, but the internal IP for the server itself. mosh works fine, but I don't know how it decides the IPs to send the UDP packets.
